### PR TITLE
VideoPlayer fixes

### PR DESCRIPTION
**Pull request by FernetMenta**
@Voyager1 623c5a7 maybe fixes dvd stills for DXVA.

### CODE REVIEW

**Comment by fritsch** [comment]
jenkins build this please

(fixes all my pain and trouble from this day and finally I can go to bed after fernet's salvation ;-))

**Comment by FernetMenta** [comment]
jenkins build this please

File: xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecFFmpeg.cpp [hunk 1]
```diff
@@ -146,6 +146,10 @@ int CDVDAudioCodecFFmpeg::Decode(uint8_t* pData, int iSize)
   av_init_packet(&avpkt);
   avpkt.data = pData;
   avpkt.size = iSize;
+  int64_t ts = (dts == DVD_NOPTS_VALUE) ? AV_NOPTS_VALUE : dts / DVD_TIME_BASE * AV_TIME_BASE;
+  avpkt.dts = ts;
+  ts = (pts == DVD_NOPTS_VALUE) ? AV_NOPTS_VALUE : pts / DVD_TIME_BASE * AV_TIME_BASE;
+  avpkt.pts = ts;
```
Comment (ace20022) [inline xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecFFmpeg.cpp]: Why the intermediate variable?

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp [hunk 2]
```diff
@@ -515,14 +515,11 @@ int CDVDVideoCodecFFmpeg::Decode(uint8_t* pData, int iSize, double dts, double p
...
-  else\
-    avpkt.ts = AV_NOPTS_VALUE
-  SET_PKT_TS(pts);
-  SET_PKT_TS(dts);
-#undef SET_PKT_TS
+  int64_t ts = (dts == DVD_NOPTS_VALUE) ? AV_NOPTS_VALUE : dts / DVD_TIME_BASE * AV_TIME_BASE;
+  avpkt.dts = ts;
+  ts = (pts == DVD_NOPTS_VALUE) ? AV_NOPTS_VALUE : pts / DVD_TIME_BASE * AV_TIME_BASE;
+  avpkt.pts = ts;
+
```
Comment (ace20022) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp]: See above

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DXVA.cpp [hunk 3]
```diff
@@ -1004,7 +1004,7 @@ int CDecoder::Decode(AVCodecContext* avctx, AVFrame* frame)
     return VC_BUFFER;
   }
   else
-    return 0;
+    return VC_BUFFER;
```
Comment (ace20022) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DXVA.cpp]: The else and the return from 1004 could be removed

File: xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecFFmpeg.cpp (hunk 1 above)
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecFFmpeg.cpp]: good question. I will drop it. thanks

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DXVA.cpp (hunk 3 above)
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DXVA.cpp]: yes, currently that would not change behaviour but mix-up things somehow. requesting a buffer for the drain path is not exactly the same thing as requesting a buffer after a pic was decoded.

**Comment by FernetMenta** [comment]
jenkins build this please

**Pull request merged**